### PR TITLE
Clear_cache inside devito

### PR DESCRIPTION
**Pull request by FabioLuporini**
You know what this is about.

~Obviously: **This PR isn't ready to be merged yet**~

### CODE REVIEW

**Comment by navjotk** [comment]
Ok this looks like a nice cleanup on the way. What is missing now?

**Comment by FabioLuporini** [comment]
> Ok this looks like a nice cleanup on the way. What is missing now?

Pretty much ready. I think it needs some beta testing, as it's a quite profound change

File: devito/types/basic.py [hunk 1]
```diff
@@ -1116,10 +1127,23 @@ class CacheManager(object):
     data is lost (and thus memory is freed).
     """
 
+    gc_ths = 3*10**8
```
Comment (georgebisbas) [inline devito/types/basic.py]: How is this value chosen?

File: examples/seismic/tutorials/02_rtm.ipynb [hunk 2]
```diff
@@ -470,7 +470,7 @@
    ],
    "source": [
     "# Run imaging loop over shots\n",
-    "from devito import Function, clear_cache\n",
+    "from devito import Function\n",
```
Comment (mloubout) [inline examples/seismic/tutorials/02_rtm.ipynb]: So nitpicking, but currently `u0` is still created once at the beginning and passed to the operator for each shot so memory overhead is still tiny without clear_cache the way the loop is implemented. Would be better to remove `u0=..., u0.data.fill` and actually see if creating a new `u` at each shots doesn't blow up memory. Same for FWI.

**Comment by FabioLuporini** [comment]
unfortunately it turns out that calling `clear_cache` at the end of `op.apply` doesn't suffice. I'm closing this PR and will open a new one as soon as it's ready

**Pull request closed without merging**